### PR TITLE
Refactor RealmSchema to remove the native handles

### LINKED ISSUE #789: Update the Schema creation logic for the new ownership semantics in Object Store
Introduced in #772

### PR DESCRIPTION
**Pull request by fealebenpae**
Closes #789

### CODE REVIEW

File: Realm.Shared/handles/SharedRealmHandle.cs [hunk 1]
```diff
@@ -166,5 +173,44 @@ public ulong GetSchemaVersion()
             nativeException.ThrowIfNecessary();
             return result;
         }
+
+        private class SchemaMarshaller
```
Comment (kristiandupont) [inline Realm.Shared/handles/SharedRealmHandle.cs]: Couple of double-l typos here..

**Comment by kristiandupont** [comment]
:+1:

File: wrappers/src/shared_realm_cs.cpp [hunk 2]
```diff
@@ -50,20 +52,71 @@ class CSharpBindingContext: public BindingContext {
...
+}
+    
 }
 }
 
 extern "C" {
-
-REALM_EXPORT void register_notify_realm_changed(NotifyRealmChangedT notifier)
+    
+    REALM_EXPORT void register_notify_realm_changed(NotifyRealmChangedT notifier)
```
Comment (AndyDentFree) [inline wrappers/src/shared_realm_cs.cpp]: Indentation typo

**Comment by AndyDentFree** [comment]
👍

**Pull request merged**